File: src/main/java/infinitespire/cards/black/Collect.java
package infinitespire.cards.black;

import com.megacrit.cardcrawl.cards.AbstractCard;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import infinitespire.abstracts.BlackCard;
import infinitespire.actions.DrawCardAndUpgradeAction;

public class Collect extends BlackCard {

	private static final String ID = "Collect";
	private static final String NAME = "Collect";
	private static final String TEXTURE = "img/infinitespire/cards/collect.png";
	private static final int COST = 1;
	private static final String DESCRIPTION = "Draw !M! cards and Upgrade them.";
	private static final int MAGIC = 3;
	
	public Collect() {
		super(ID, NAME, TEXTURE, COST, DESCRIPTION, CardType.SKILL, CardTarget.SELF);
		this.baseMagicNumber = MAGIC;
		this.magicNumber = MAGIC;
	}
	
	
	@Override
	public AbstractCard makeCopy() {
		return new Collect();
	}

	@Override
	public void upgrade() {
		if(!upgraded) {
			this.upgradeName();
			this.upgradeMagicNumber(1);
		}
	}

	@Override
	public void use(AbstractPlayer p, AbstractMonster m) {
		AbstractDungeon.actionManager.addToBottom(new DrawCardAndUpgradeAction(p, this.magicNumber));
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/cards/black/Collect.kt
package com.blanktheevil.infinitespire.cards.black

import com.blanktheevil.infinitespire.actions.DrawCardAndUpgrade
import com.blanktheevil.infinitespire.cards.utils.CardBuilder
import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.vfx.FinalStrikeVfx
import com.blanktheevil.infinitespire.vfx.UltimateFormVfx
import com.megacrit.cardcrawl.actions.animations.VFXAction

class Collect : BlackCard(BUILDER) {
  companion object {
    val ID = "Collect".makeID()
    private const val MAGIC = 3
    private const val UPG_MAGIC = 1
    private val BUILDER = CardBuilder(ID)
      .img("collect.png")
      .cost(0)
      .skill()
      .self()
      .init {
        baseMagicNumber = MAGIC
        magicNumber = MAGIC
      }
      .upgr {
        upgradeMagicNumber(UPG_MAGIC)
      }
      .use { p, _ ->
        addToBot(DrawCardAndUpgrade(p!!, magicNumber))
      }
  }
}